### PR TITLE
Custom purposes i18n

### PR DESCRIPTION
**Pull request by anotherminh**
## Related Issues

- Part of [T-19832](https://transcend.height.app/T-19832) Allow internationalization for custom purposes

## Security Implications

_[none]_

## System Availability

_[none]_

### CODE REVIEW

File: src/hooks/useGetPurposeMessageKeys.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}): PurposeMessages => {
+  const purposeToMessageKey: Record<string, DefinedMessage> = useMemo(
+    () =>
+      // the purpose type is unique for the bundle
+      Object.keys(consentSelection ?? {}).reduce(
+        (allMessages, purposeType) => {
+          if (allMessages[purposeType]) {
+            return allMessages;
+          }
+          const customPurposeMessageLabel = `purpose.${purposeType}`;
```
Comment (anotherminh) [inline src/hooks/useGetPurposeMessageKeys.ts:28]: purposeType is guaranteed to be unique and immutable, so I'd rather use this than the uuid

File: src/hooks/useGetPurposeMessageKeys.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            description: `Translatable name for custom purpose '${purposeType}'`,
+          } as DefinedMessage,
+        };
+      }, defaultPurposeToMessageKey as Record<string, DefinedMessage>),
+    [consentSelection, defaultPurposeToMessageKey],
+  );
+  const orderOfPurposes = useMemo(
+    () => Object.keys(purposeToMessageKey),
+    [purposeToMessageKey],
+  );
```
Comment (MadDataScience) [inline src/hooks/useGetPurposeMessageKeys.ts:47]: Does this mean customers will be able to customize the order or purposes?

**Review by MadDataScience — DISMISSED** [review]
This is exciting

**Review by michaelfarrell76 — DISMISSED** [review]
nice!

File: src/hooks/useGetPurposeMessageKeys.ts (hunk 2 above)
Comment (michaelfarrell76) [reply src/hooks/useGetPurposeMessageKeys.ts:47]: the existing order of purposes is on purpose - would be great to maintain that if possible!

Comment (anotherminh) [reply src/hooks/useGetPurposeMessageKeys.ts:47]: no haha we just want to make sure the order is deterministic here

Comment (anotherminh) [reply src/hooks/useGetPurposeMessageKeys.ts:47]: @michaelfarrell76 ohhh yes i see - i'll update the logic to keep that in place

Comment (michaelfarrell76) [reply src/hooks/useGetPurposeMessageKeys.ts:47]: would be nice to support this in the platform eventually, but i get thats harder

Comment (anotherminh) [reply src/hooks/useGetPurposeMessageKeys.ts:47]: Screenshot 2023-08-10 at 4 16 53 PM

done

**Review by michaelfarrell76 — APPROVED** [review]

**Comment by michaelfarrell76** [comment]
@anotherminh is this good to merge? im about to make a pr to this repo

**Pull request merged**